nexus/db-queries/src/transaction_retry.rs

```diff
@@ -151,14 +182,13 @@ impl RetryHelper {
         return inner.attempts < MAX_RETRY_ATTEMPTS;
     }
 
-    /// Converts this function to a retryable callback that can be used from
-    /// "transaction_async_with_retry".
-    pub(crate) fn as_callback(
-        self,
+    // Converts this function to a retryable callback that can be used from
+    // "transaction_async_with_retry".
+    fn as_callback(
+        self: Arc<Self>,
     ) -> impl Fn() -> futures::future::BoxFuture<'static, bool> {
-        let r = Arc::new(self);
         move || {
-            let r = r.clone();
+            let r = self.clone();
```

Comment:
Is this clone still necessary?

Reply:
Without the clone:

```
error[E0507]: cannot move out of `self`, a captured variable in an `Fn` closure
   --> nexus/db-queries/src/transaction_retry.rs:191:21
    |
188 |         self: Arc<Self>,
    |         ---- captured outer variable
189 |     ) -> impl Fn() -> futures::future::BoxFuture<'static, bool> {
190 |         move || {
    |         ------- captured by this `Fn` closure
191 |             let r = self;
    |                     ^^^^ move occurs because `self` has type `Arc<RetryHelper>`, which does not implement the `Copy` trait
    |
help: consider borrowing here
    |
191 |             let r = &self;
    |                     +

For more information about this error, try `rustc --explain E0507`.
error: could not compile `nexus-db-queries` (lib) due to previous error
```

I don't actually want to give a reference there, because I want to "move the Arc into the closure", but for whatever reason, that `move` isn't letting rust know to move the arc into the closure...

Reply:
Oh, because the return value is an `impl Fn` which can be called many times. Never mind

Reply:
Ah, yeah, it has to be -- it's retryable, since that's the whole point of this.